### DISCUSSION TITLE
Request: add examples of using r2d2 with Diesel

### DISCUSSION THREAD

**Discussion by DeflateAwning** [diesel-rs/diesel#3984] [Ideas]
It seems like r2d2 supports many more databases than Diesel, and my understanding is that it can act as a "middleware" between diesel and those databases (e.g., firebird, etc.). 

Would be awesome if you could an example of how to use this!

**Comment by weiznich** [comment]
r2d2 is a standalone project. It's used as dependency of diesel, but diesel only provides integration for diesel based connections. Everything else is implemented in different crates. You can find the relevant documentation in the [main r2d2 crate documentation](https://docs.rs/r2d2/latest/r2d2/) or in any of the helper crates that provide support for specific connection types (see the relevant [table in the README](https://github.com/sfackler/r2d2?tab=readme-ov-file#r2d2)).

That's nothing that belongs to diesel itself.
